sycl/doc/extensions/experimental/sycl_ext_intel_esimd/sycl_ext_intel_esimd.md

```diff
@@ -511,6 +511,114 @@ ESIMD_PRIVATE ESIMD_REGISTER(32) simd<int, 16> vc;
 ```
 <br>
 
+### `__regcall` Calling convention.
+
+ESIMD supports `__regcall` calling convention (CC) in addition to the default
+SPIR CC. This makes compiler try generate more efficient calls where arguments
+of aggregate types (classes, structs, unions) are passed and values returned via
+registers rather than memory. This matters most for external functions linked on
+binary level, such as functions called via `invoke_simd`. Arguments and return
+values ("ARV") are still passed or returned ("communicated") via a pointer if
+their type is either of the following:
+- a class or struct with deleted copy constructor
+- an empty class or struct
+- a class or struct ending with a flexible array member. For example:
+`class A { int x[]; }`
+
+ARVs of all other aggregate types are communicated by value or "per-field". Some
+fields can be replaced with 1 or 2 integer elements with total size being equal
+or exceeding the total size of fields. The rules for communicating ARVs of these
+types are part of the SPIR-V level function call ABI, and are described below.
+This part of the ABI is defined in terms of LLVM IR types - it basically
+tells how a specific source aggregate type is represented in resulting LLVM IR
+when it (the type) is part of a signature of a function with linkage defined.
+
+Compiler uses aggregate type "unwrapping process" for communicating ARVs.
+Unwarapping a structure with a single field results in the unwrapped type of
+that field, so unwrapping is a recursive process. Unwrapped primitive type is
+the primitive type itself. Structures with pointer fields are not unwrapped.
+For example, unwrapping `Y` defined as
+```cpp
+struct X { int x; };
+struct Y { X x; };
+```
+results in `i32`. Unwrapping `C4` defind as
+```cpp
+struct A4 { char x; };
+struct B4 { A4 a; };
+struct C4 {
+  B4 b;
+  int *ptr;
+};
+```
+results in { `%struct.B4`, `i32 addrspace(4)*` } pair of types. Thus,
+unwraping can result in a set of a structure, primitive or pointer types -
+the "unwrapped type set".
+
+- If the unwrapped type set has only primitive types, then compiler will "merge"
+  the resulting types if their total size is less or equal to 8 bytes. The total
+  size is calculated as `sizeof(<top aggregate type>)`, and structure field
+  alignment rules can make it greater then the simple sum of `sizeof` of all
+  the types resulted from unwrapping. [Total size] to [merged type]
+  correspondence is as follows:
+    * 1-2 bytes - short
+    * 3-4 bytes - int
+    * 5-8 bytes - array of 2 ints
+  Floating point types are not merged. Structure field alignment rules can
+  increase the calculated size compared to simple sum of `sizeof` of all the
+  types. If the total size exceeds 8, then:
```

Comment:
> Structure field alignment rules can increase the calculated size compared to simple sum of `sizeof` of all the types.

I think this can be removed since it is mentioned above when describing how size is calculated.

Reply:
OK